src/resources/favorites.ts

```diff
@@ -121,36 +127,44 @@ async function fetchMetadata(addresses: string[], network = Network.mainnet) {
 /**
  * Refreshes the metadata associated with all favorites.
  */
-export async function refreshFavorites() {
+export async function refreshFavorites(chainId = ChainId.mainnet) {
   const favorites = Object.keys(queryClient.getQueryData(favoritesQueryKey) ?? DEFAULT);
-  const updatedMetadata = await fetchMetadata(favorites);
+  const updatedMetadata = await fetchMetadata(favorites, chainId);
   return updatedMetadata;
 }
 
-export async function toggleFavorite(address: string) {
-  const favorites = Object.keys(queryClient.getQueryData(favoritesQueryKey) ?? []);
-  const lowercasedAddress = address.toLowerCase();
-  let updatedFavorites;
-  if (favorites.includes(lowercasedAddress)) {
-    updatedFavorites = without(favorites, lowercasedAddress);
+/**
+ * Toggles the favorite status of the given address. Performs a fetch to refresh the metadata of the
+ * given address if it is not already favorited. If the address is already favorited, the favorite status
+ * is removed from the query data.
+ * @param address - The address to toggle the favorite status of.
+ * @param chainId - The chain id of the network to toggle the favorite status of @default ChainId.mainnet
+ */
+export async function toggleFavorite(address: string, chainId = ChainId.mainnet) {
+  const favorites = queryClient.getQueryData<Record<EthereumAddress, RainbowToken>>(favoritesQueryKey);
+  const lowercasedAddress = address.toLowerCase() as EthereumAddress;
+  if (Object.keys(favorites || {}).includes(lowercasedAddress)) {
+    queryClient.setQueryData(favoritesQueryKey, omit(favorites, lowercasedAddress));
   } else {
-    updatedFavorites = [...favorites, lowercasedAddress];
+    const metadata = await fetchMetadata([lowercasedAddress], chainId);
+    queryClient.setQueryData(favoritesQueryKey, {
+      ...favorites,
+      ...metadata,
+    });
   }
-  const metadata = await fetchMetadata(updatedFavorites);
-
-  queryClient.setQueryData(favoritesQueryKey, metadata);
 }
 
 /**
  * Returns `favorites`, an array of favorited addresses, as well as `favoritesMetadata`, a map of these
  * addresses to their corresponding `RainbowToken`. These values are cached in AsyncStorage and is only
  * modified/updated when `toggleFavorite` or `refreshFavorites` is called.
  */
-export function useFavorites(): {
+export function useFavorites(chainId = ChainId.mainnet): {
   favorites: string[];
   favoritesMetadata: Record<EthereumAddress, RainbowToken>;
 } {
-  const query = useQuery<Record<EthereumAddress, RainbowToken>>(favoritesQueryKey, refreshFavorites, {
+  const queryFn = () => refreshFavorites(chainId);
+  const query = useQuery<Record<EthereumAddress, RainbowToken>>(favoritesQueryKey, queryFn, {
```

Comment:
I think `chainId` should be in the `queryKey`

Reply:
I agree but we use these favorites elsewhere in the app so I have to keep the functionality the same here.